crates/volta-core/src/tool/package/mod.rs

```diff
@@ -35,7 +35,7 @@ pub struct Package {
 
 impl Package {
     pub fn new(name: String, version: VersionSpec) -> Fallible<Self> {
-        let staging = setup_staging_directory(PackageManager::Npm)?;
+        let staging = setup_staging_directory(PackageManager::Npm, false /* needs_scope */)?;
```

Comment:
The need for the comment is… suggestive… of the high utility we'd get from refactoring this to use an enum (_a la_ [Booleans and Enums](https://thoughtbot.com/blog/booleans-and-enums)):

```rust
#[derive(PartialEq)]
enum NeedsScope {
    Yes,
    No,
}
```

Because of this kind of readability/comprehensibility issue, I've actually gotten to the point where I'll just straight-up introduce new boolean enums anywhere it helps. It makes a *very* small difference in the size of the compiled binary and otherwise no difference perf-wise, so it seems very much worth the tradeoff to me.

```suggestion
        let staging = setup_staging_directory(PackageManager::Npm, NeedsScope::No)?;
```

Reply:
I definitely went back and forth on this, I _didn't_ like using a boolean, so yeah I think a boolean Enum is worth it!